lib/Thruk/Authentication/User.pm

```diff
@@ -222,8 +222,22 @@ sub get_dynamic_roles {
     }
 
     if(!defined $can_submit_commands) {
-        $can_submit_commands = $c->config->{'can_submit_commands'} || 0;
-    }
+	if (defined $c->config->{'can_submit_commands'}) {
+		# honor global Thruk configuration
+	        $can_submit_commands = $c->config->{'can_submit_commands'};
+	} else {
+		# stay secure by default
+	        $can_submit_commands = 0;
+	};
+    }
+
+    if (defined $c->config->{'User'}->{$c->user->{'username'}}->{'can_submit_commands'}) {
```

Comment:
this looks like it should not be necessary. `$c->config` should have the user specific config already merged into, except usually also groups would be honored if they exist. So if that does not work as intended right now, then this should be fixed instead.

Reply:
without the patch this very strange behavior is occuring that logged-in users can sometimes execute commands, sometimes not. With the patch it's working fine...No clue where the root cause is located for that issue (potentially a race condition somewhere) - but until found I have to patch the file after every update.

Reply:
the problem is, when using user/group config overrides, the current implementation overwrites the global configuration and resets it after the request. Now, if can imagine, if so some reason the reset does not happen because the request exits early, then those temporary adjustments become permanent. This is not ideal. I will see if i can invert that, so
that the global configuration does not have to changed.